### DISCUSSION TITLE
Does pv.start_xvfb() cause memory leakage during multiple API calls?

### DISCUSSION THREAD

**Discussion by jais001** [pyvista/pyvista#6560] [Q&A] [answered]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```python
Does this cause memory leakage during multiple API calls? How does xvfb stop?

class Visualize:
    """Visualize Service"""

    def __init__(self) -> None:
        pv.start_xvfb()
```

### Description

I have an application that uses Pyvista for rendering. I have dockerized the application. But after calling the API that uses Pyvista I have got an error that states that:

`Try starting a virtual frame buffer with xvfb, or using `pyvista.start_xvfb() warnings.warn( 2024-08-22 10:49:54.850 ( 924.183s) [ 79CA01C006C0]vtkXOpenGLRenderWindow.:456 ERR| vtkXOpenGLRenderWindow (0x79c9f01f32b0): bad X server connection. DISPLAY= ERROR:root:bad X server connection. DISPLAY= `

I have tried the bash script execution in my docker file after referring to the pyvista documentation:
https://docs.pyvista.org/getting-started/installation.html#running-on-remote-servers

But still, I got the error, so I had to add pv.start_xvfb() for the API to work. I haven't seen anywhere in the documentation to close xvfb, so I would like to know if it causes memory leakage during multiple API calls. how does itxvfb stop? or are there any other workarounds, because the bash script did not work while executing in my Dockerfile?

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Fri Aug 23 10:55:17 2024 IST

                OS : Linux (Ubuntu 20.04)
            CPU(s) : 12
           Machine : x86_64
      Architecture : 64bit
               RAM : 15.5 GiB
       Environment : Python
       File system : ext4
        GPU Vendor : NVIDIA Corporation
      GPU Renderer : NVIDIA GeForce GTX 1650/PCIe/SSE2
       GPU Version : 4.5.0 NVIDIA 555.42.06
  MathText Support : True

  Python 3.11.0 (main, Mar  1 2023, 18:26:19) [GCC 11.2.0]

           pyvista : 0.44.1
               vtk : 9.3.1
             numpy : 1.26.4
        matplotlib : 3.9.2
            scooby : 0.10.0
             pooch : 1.8.2
            pillow : 10.4.0
             scipy : 1.14.0
              tqdm : 4.66.5
             trame : 3.6.3
      trame_client : 3.2.5
      trame_server : 3.1.0
         trame_vtk : 2.8.10
     trame_vuetify : 2.6.2
      nest_asyncio : 1.6.0
--------------------------------------------------------------------------------
```

**Comment by tkoyama010** [comment, marked as answer]
Thanks for your post. `start_xvfb` is deprecated in #6500. Please use `vtk-osmesa` instead.
https://dev.pyvista.org/user-guide/jupyter/trame.html#other-considerations

**Comment by jais001** [comment]
Hi @tkoyama010, thanks for the info. Now it is working fine...
